src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -28,34 +28,109 @@
 namespace esp {
 namespace physics {
 
-BulletRigidObject::BulletRigidObject(scene::SceneNode* rigidBodyNode)
-    : RigidObject{rigidBodyNode}, MotionState(*rigidBodyNode){};
-
-bool BulletRigidObject::initializeScene(
+BulletRigidObject::BulletRigidObject(
+    scene::SceneNode* rigidBodyNode,
+    std::shared_ptr<btMultiBodyDynamicsWorld> bWorld)
+    : RigidObject{rigidBodyNode},
+      MotionState(*rigidBodyNode),
+      bWorld_(bWorld) {}
+
+bool BulletRigidObject::initializeSceneFinalize(
+    const assets::ResourceManager* resMgr,
```

Comment:
```suggestion
    const assets::ResourceManager& resMgr,
```

instead (and same for other functions)? This function accesses the pointer without checking it's non-null and the documentation doesn't explicitly say that it's allowed to be null, so I assume it's not, and thus I think it would be clearer to enforce this by accepting a reference.

Reply:
the reason i am using a pointer is that the calling manager has it as a pointer already.  just passing what was there already.

Reply:
Perhaps they should be friended and then made private?

Reply:
In my case I'm always trying to move the responsibility of null checking to the bottom of the stack as much as possible, and since here it clearly always needs a non-null pointer, I would make these (and `initializeScene()`) accept references. Is there a piece of code clearly responsible for creating the pointer? Or can it be references all the way down?

Reply:
We could refactor to `ResourceManager` reference in `PhysicsManager` and pass that along the full path. I don't see any problem with that. 👍

Reply:
Let's call this a follow-up refactor PR.